spec/lib/extensions/brakeman_fingerprint_patch_spec.rb

```diff
@@ -41,127 +56,219 @@ def location = {}
             :absolute => "/Users/user/dev/manageiq-ui-classic/app/controllers/application_controller.rb"
           )
         end
 
+        let(:other_engine_path) { "/Users/user/.gem/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f" }
+        let(:other_engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "../../.gem/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f/app/controllers/api/base_controller.rb",
+            :absolute => "/Users/user/.gem/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f/app/controllers/api/base_controller.rb"
+          )
+        end
+
         include_examples "handles brakeman pathing"
       end
 
-      context "which is inside of the core repo" do
-        let(:engine_path) { "/Users/user/dev/manageiq/plugins/manageiq-ui-classic/" }
+      context "which is inside of the core repo but some other engine is a git-based gem" do
+        let(:engine_path) { "/Users/user/dev/manageiq/plugins/manageiq-ui-classic" }
         let(:engine_issue) do
           instance_double("Brakeman::FilePath",
             :relative => "plugins/manageiq-ui-classic/app/controllers/application_controller.rb",
             :absolute => "/Users/user/dev/manageiq/plugins/manageiq-ui-classic/app/controllers/application_controller.rb"
           )
         end
 
+        let(:other_engine_path) { "/Users/user/.gem/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f" }
+        let(:other_engine_issue) do
+          instance_double("Brakeman::FilePath",
+            :relative => "../../.gem/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f/app/controllers/api/base_controller.rb",
+            :absolute => "/Users/user/.gem/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f/app/controllers/api/base_controller.rb"
+          )
+        end
+
         include_examples "handles brakeman pathing"
       end
     end
 
     context "with a cloned spec/manageiq dir" do # This is also the CI case for a plugin
-      let(:core_path) { "/Users/user/dev/manageiq-ui-classic/spec/manageiq/" }
+      let(:core_path) { "/Users/user/dev/manageiq-ui-classic/spec/manageiq" }
       let(:core_issue) do
         instance_double("Brakeman::FilePath",
           :relative => "lib/ansible/runner.rb",
           :absolute => "/Users/user/dev/manageiq-ui-classic/spec/manageiq/lib/ansible/runner.rb"
         )
       end
 
-      let(:engine_path) { "/Users/user/dev/manageiq-ui-classic/" }
+      let(:engine_path) { "/Users/user/dev/manageiq-ui-classic" }
       let(:engine_issue) do
         instance_double("Brakeman::FilePath",
           :relative => "../../app/controllers/application_controller.rb",
           :absolute => "/Users/user/dev/manageiq-ui-classic/app/controllers/application_controller.rb"
         )
       end
 
+      let(:other_engine_path) { "/Users/user/dev/manageiq-ui-classic/vendor/bundle/ruby/3.1.5/bundler/gems/manageiq-api-968f4c010f2f" }
```

Comment:
This is really the important test - all of the other ones are just adding that as an extra case